rpc/src/main/scala/WorkerServer.scala

```diff
@@ -167,10 +161,42 @@ class WorkerServer(
     }
 
     override def exchange(request: ExchangeRequest): Future[ExchangeReply] = {
-      val futures =
-        request.workers.map(_ => Future {}(executionContext))
+      val promise = Promise[ExchangeReply]
+
+      Future {
+        logger.debug(s"[WorkerServer] Exchange ($request)")
+
+        val workers = toWorkerMetadata(request.workers)
+
+        inputFiles.foreach(path => {
```

Comment:
I think this 'inputFiles' is not valid in exchange stage because in partition function(at WorkerServer.scala), deletes file. 
So my suggestion is...
1. Add in 'val partitionedFiles' in class WorkerService which can be used for saving newly added files.
2. Add 'partitionPath' saving logic in partition funciton(at WorkerServer.scala).
3. Replace 'inputFiles' to 'partitionedFiles' in exchange funciton.

Reply:
![image](https://github.com/betarixm/434project/assets/110564003/e608b055-80ff-42d1-95a4-8097e8e4c4bb)

Reply:
Note that `inputFiles` is defined with `def`, not a `val`.

```scala
private def inputFiles: Array[File] = inputDirectories.flatMap(_.files)
```

Reply:
My Bad. :)